graph-gateway/src/client_query.rs

```diff
@@ -113,13 +111,19 @@ pub async fn handle_query(
     tracing::info!(deployments = ?deployments.iter().map(|d| d.id).collect::<Vec<_>>());
 
     // Check authorization for the resolved deployments
-    if !auth.are_deployments_authorized(&deployments) {
+    let resolved_deployments_ids = deployments.iter().map(|d| d.id).collect::<Vec<_>>();
```

Comment:
we should move the log event (above this) down do avoid collecting IDs into a vec twice unnecessarily